src/components/Card/Card.tsx

```diff
@@ -0,0 +1,17 @@
+import useNavigateToPage from '../../hooks/useNavigate';
+import { ICardData } from '../../interfaces/search-result.interface';
+import styles from './Card.module.css';
+
+export default function Card({ title, images, mal_id }: ICardData) {
+  const { navigateToCard } = useNavigateToPage();
+
+  return (
+    <div
+      className={styles.container}
+      onClick={() => navigateToCard(`/${mal_id}`)}
```

Comment:
```tsx
<Link to=`/someroute`>
...your code
<Link>
```
?

Reply:
I don’t think the <Link> works correctly in my case, as I want it to display the query parameters in the address bar when navigating to the detailed page using the id. I have implemented this in a custom hook useNavigate, but perhaps there is a more elegant way to achieve this

Reply:
ok, it's not a big deal